girder/models/api_key.py

```diff
@@ -50,8 +50,14 @@ def validate(self, doc):
             if not isinstance(doc['scope'], (list, tuple)):
                 raise ValidationException('Scope must be a list, or None.')
             if not doc['scope']:
-                raise ValidationException(
-                    'Custom scope list must not be empty.')
+                raise ValidationException('Custom scope list must not be empty.')
+
+            # Ensure only registered scopes are being set
+            admin = self.model('user').load(doc['userId'], force=True)['admin']
+            scopes = TokenScope.scopeIds(admin)
+            unknown = set(doc['scope']) - scopes
```

Comment:
@zachmullen Can we have a more descriptive name for unknown? Maybe unknownScope?